diff -rupN backports-3.11-rc3-1.orig/compat/compat-3.0.c backports-3.11-rc3-1.new/compat/compat-3.0.c
--- backports-3.11-rc3-1.orig/compat/compat-3.0.c	2013-07-30 11:58:54.000000000 +0200
+++ backports-3.11-rc3-1.new/compat/compat-3.0.c	2013-09-20 18:12:08.995515332 +0200
@@ -12,11 +12,15 @@
 #include <linux/compat.h>
 #include <linux/if_ether.h>
 
+
 #if (RHEL_RELEASE_CODE < RHEL_RELEASE_VERSION(6,4))
 /* This pulls-in a lot of non-exported symbol backports
  * on kernels older than 2.6.32. There's no harm for not
  * making this available on kernels < 2.6.32. */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,32))
+// leJOS
+#if 0
+// This function does not seem to be defined in the Lego kernel. 
+//#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,32))
 #include <linux/pagemap.h>
 #include <linux/shmem_fs.h>
 
diff -rupN backports-3.11-rc3-1.orig/compat/compat_atomic.c backports-3.11-rc3-1.new/compat/compat_atomic.c
--- backports-3.11-rc3-1.orig/compat/compat_atomic.c	2013-07-30 11:58:54.000000000 +0200
+++ backports-3.11-rc3-1.new/compat/compat_atomic.c	2013-09-20 18:04:29.403492119 +0200
@@ -1,7 +1,11 @@
 #include <linux/spinlock.h>
 #include <linux/module.h>
+// leJOS
+#include <asm-generic/atomic64.h>
 
-#if !((LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,31)) && (defined(CONFIG_UML) || defined(CONFIG_X86))) && !((LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,33)) && defined(CONFIG_ARM) && !defined(CONFIG_GENERIC_ATOMIC64))
+//#if !((LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,31)) && (defined(CONFIG_UML) || defined(CONFIG_X86))) && !((LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,33)) && defined(CONFIG_ARM) && !defined(CONFIG_GENERIC_ATOMIC64))
+#if 1
+// leJOS We need this code on ARM
 
 static DEFINE_SPINLOCK(lock);
 
diff -rupN backports-3.11-rc3-1.orig/net/mac80211/key.h backports-3.11-rc3-1.new/net/mac80211/key.h
--- backports-3.11-rc3-1.orig/net/mac80211/key.h	2013-07-30 11:58:53.000000000 +0200
+++ backports-3.11-rc3-1.new/net/mac80211/key.h	2013-09-20 18:07:26.551501066 +0200
@@ -15,6 +15,8 @@
 #include <linux/crypto.h>
 #include <linux/rcupdate.h>
 #include <net/mac80211.h>
+// leJOS
+#include <asm-generic/atomic64.h>
 
 #define NUM_DEFAULT_KEYS 4
 #define NUM_DEFAULT_MGMT_KEYS 2
